core/src/main/java/com/graphhopper/storage/index/LocationIndexTree.java

```diff
@@ -564,6 +572,133 @@ protected boolean check(int node, double normedDist, int wayIndex, EdgeIteratorS
         return closestMatch;
     }
 
+    /**
+     * Returns all edges that are within the specified radius around the queried position.
+     *
+     * @param queryLat latitude to search from
+     * @param queryLon longitude to search from
+     * @param edgeFilter only include edges from edgeFilter
+     * @param radius include all results within this radius (in meters) - assuming it's not
+     *        bounded by maxIterations.
+     * @param maxIterations if not null, search only a square of (2 * maxIterations - 1) tiles
```

Comment:
I don't get the purpose of maxIterations and radius. 

> include all results within this radius (in meters) - assuming it's not bounded by maxIterations.

How can I bound the radius with maxIterations, when maxIterations says:

> Note that it is possible that the search radius exceeds this square - in this case, an exception will be thrown.

To me this sounds contradicting. You can bound the radius with maxIterations, but maxIterations has to be at least as big as the radius.

So maxIterations can increase the radius? But why bothering, I could just increase the radius.

BTW: At this point haven't really looked at the code, I just stumbled over the JavaDoc.

**Edit:**
Ok, I am finished with reviewing the code and still don't understand maxIterations.
Is it necessary to have maxIterations? If yes why? And if yes, wouldn't it be easier to have a wrapper method that allows calling this method without maxIterations?